### DISCUSSION TITLE
App support with different types of proxies

### DISCUSSION THREAD

**Discussion by Vicio59** [engageub/InternetIncome#87] [Q&A]
Hi engageub, congratulations and thanks for your work. I have some questions about using residential ISP proxies.
Will all apps work if I use a proxy that supports TCP or is it mandatory to have a proxy that supports UDP?
Which ones will work and which ones won't?
Same problem with proxies that don't support DNS which apps will work and which won't? Thanks in advance, regards

**Comment by engageub** [comment]
Hello @Vicio59,
Based on the usage of apps, except Mysterium, all other apps use UDP for DNS queries. 
It is recommended that you use socks5 with UDP support to avoid DNS leaks. However, it is not an issue as of now. 

Thank you

**Reply by Vicio59** [reply]
Thanks for the quick answer, sorry if I ask you another question. I read the providers you recommend here: https://github.com/engageub/InternetIncome/wiki/Proxies-and-Hosting-Providers , which of these support udp and DNS or have you seen that they work well with apps? or alternatively if you know of any  good provider that is not listed here. Thank you

**Reply by engageub** [reply]
Hi,
As of now you can use the apps without UDP/DNS support on proxies.
Among the providers, proxyfish supports UDP over socks5 proxy.

Thank you
